### DISCUSSION TITLE
Display the spell graphic in the Shrines and Pyramid windows when the hero who visits them already has the spell

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#4618] [Ideas]
I think it would be interesting to display the spell graphic in the Shrines and Pyramid windows between the first part of the text and the end of the text, when the hero who visits them already has the spell.

In the example, between:

"...spell - 'Teleport'."

And:

"Unfortunately, you already..."

And also, allow the player to right click on the spell graphic to display the spell information window.

In fheroes2:

![Spell cost 005](https://user-images.githubusercontent.com/43583994/141355702-5b324990-c07f-4b8f-9019-55a40cdc30d1.png)

In original game:
In fheroes2, currently these windows are not displayed correctly: https://github.com/ihhub/fheroes2/issues/4615.

![Pyramid message 03](https://user-images.githubusercontent.com/43583994/141346278-736df50d-6032-4bf5-8dfc-bf70eae701c9.png)

![Pyramid message 05](https://user-images.githubusercontent.com/43583994/141346291-747ab844-0f34-4567-9f10-ed89e9168c7a.png)

**Comment by Branikolog** [comment]
Hi, @LeHerosInconnu !
UI of the original "Heroes" game often works like this: the reward a hero had received appears on a screen: spell icon in the case above,  gold from water wheel or an artifact being gathered from a ground. Game doesn't show gold pile, if water wheel is empty. This icons being drawn in a window on success action make these windows more distinguishable, from a case, when an action brought you nothing.

What do you think if we just highlight spell name within the text with **different color**? So player won't need to read whole text to find out, what was the spell, but at the same time he clearly understands, that he wasn't succeeded in learning it, as there's no spell icon in the window.

**Reply by LeHerosInconnu** [reply]
Hello @Branikolog,

> Hi, @LeHerosInconnu ! UI of the original "Heroes" game often works like this: the reward a hero had received appears on a screen: spell icon in the case above, gold from water wheel or an artifact being gathered from a ground. Game doesn't show gold pile, if water wheel is empty. This icons being drawn in a window on success action make these windows more distinguishable, from a case, when an action brought you nothing.

This is not really the same case for the Water Wheel. :)
And I thought about that too.
But I figured that since there was extra text displayed after the added spell graphic, the player could very quickly integrate that, even visually without having to read the text below the added spell graphic.
No text below the graphic, it's a "gain", text below the graphic, it's just information.

I plan to extend this same principle to the Witch Hut for the secondary skill, etc.

In fheroes2:

![Witch Hut 001](https://user-images.githubusercontent.com/43583994/141473225-06ef36bd-33a8-4148-8f21-76358f236b03.png)

![Witch Hut 002](https://user-images.githubusercontent.com/43583994/141473245-e7b1a4eb-09df-4b18-8eff-4bfbd8e2e12e.png)

> What do you think if we just highlight spell name within the text with **different color**? So player won't need to read whole text to find out, what was the spell, but at the same time he clearly understands, that he wasn't succeeded in learning it, as there's no spell icon in the window.

I think that since the graphic is available, it would be a shame not to use it. :)
And with the added graphic it would not even be necessary to read the name of the spell, the player can understand the information given in the window graphically (visually), simply by the position of the graphic in the window (at the bottom, or in the center (between two text blocks)).
And it would also be a plus to be able to right click on the graphic to get information about it directly in this window, which is a "natural player action" in the game interface.
Also, the name of the spell (and other) can, in addition to the added graphic, also be written in another color, this is a good idea. :)

**Reply by Branikolog** [reply]
> And it would also be a plus to be able to right click on the graphic to get information about it directly in this window, which is a "natural player action" in the game interface.

Definitely yes.

> I think that since the graphic is available, it would be a shame not to use it. :)

Well, hard to argue with that...

> And with the added graphic it would not even be necessary to read the name of the spell, the player can understand the information given in the window graphically (visually), simply by the position of the graphic in the window (at the bottom, or in the center (between two text blocks)).

Actually text in this window lacks enough contrast. In my point of view presence/absence of spell icon in this window is much more distinguishable, since players often skip this window in less than a second. 
My second point - not showing icon is a common implementation of this logic in HoMM series and players are really used to it.

Just my thoughts...

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> > And with the added graphic it would not even be necessary to read the name of the spell, the player can understand the information given in the window graphically (visually), simply by the position of the graphic in the window (at the bottom, or in the center (between two text blocks)).
> 
> Actually text in this window lacks enough contrast. In my point of view presence/absence of spell icon in this window is much more distinguishable, since players often skip this window in less than a second. My second point - not showing icon is a common implementation of this logic in HoMM series and players are really used to it.
> 
> Just my thoughts...

If the player closes the window in less than a second, it doesn't really matter what is inside the window. :D

If the player quickly looks at the window and sees the graphic at the bottom of the window and closes the window in less than a second, he will quickly understand and integrate that when the graphic is between two blocks of text in the middle of the window it is just information and not a reward.

Also, a different sound can be played if the player already knows the spell (or other) or not. :)

**Reply by Branikolog** [reply]
> If the player quickly looks at the window and sees the graphic at the bottom of the window and closes the window in less than a second, he will quickly understand and integrate that when the graphic is between two blocks of text in the middle of the window it is just information and not a reward.

A colorful icon in this window and its absence is the first and the main thing, what each player notices and is focused on. Text - is just a text. Players rarely pay attention to that small pieces of letters, no matter where they're located within a window. :) You know, There're still tons of players, who don't know how to expand the game into full screen, considering the fact we've added the description of F4 button on the first start right in the middle of the screen. :D

**Reply by zenseii** [reply]
I think marking the Skill or whatever of the dialog text with a yellow color is a great idea for readability. While adding the icon in both the window where you get it and where you don't, to me is a very big change and might not be well received among veteran players. What if you could gray out the color of the icon if you don't get it?

**Reply by Branikolog** [reply]
> What if you could gray out the color of the icon if you don't get it?

Nice idea!
But I'm afraid in this particular case this could be not so obvious on some icons. For example "Slow" spell is mostly black with thin yellow edges...  Also some icons are already gray ("Stone skin")

**Reply by Branikolog** [reply]
Well, if we talk about suggestions, I can offer something too, from my side. But, actually, I'm OK with original implementation with yellow spell name. 
What if we show spell icon, hero has learned, placed in a scroll, like that, we have in a mage guild? 
In a case hero cannot learn spell - we show rolled up (closed) scroll, the same to scrolls of mage levels not being built in a mage guild? :)

**Reply by zenseii** [reply]
That sounds like an even better idea and something that could have natively been in the game.

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> What if we show spell icon, hero has learned, placed in a scroll, like that, we have in a mage guild? In a case hero cannot learn spell - we show rolled up (closed) scroll, the same to scrolls of mage levels not being built in a mage guild? :)

The goal is to have knowledge of the spell without having to read, and this whether the spell is already learned or not.
Now, if the spell scroll is closed, you don't see anything, and you have to read the text, and look for the name of the spell.

**Comment by LeHerosInconnu** [comment]
This is planned in my mod: https://github.com/ihhub/fheroes2/discussions/10092, https://github.com/ihhub/fheroes2/discussions/10093, https://github.com/ihhub/fheroes2/discussions/10095.
I close the discussion.
